doc/migrations/6.2.1_x.x.x.md

```diff
@@ -2,3 +2,94 @@
 
 Hook names have been unified by removing the `_hook` prefix from the few hooks which used it,
 e.g. `offline_message_hook` is now called `offline_message`. This change affects the hook metric names as well.
+
+Some hooks have been removed:
+
+ - `xmpp_stanza_dropped`,
+ - `privacy_list_push`,
+ - `xmpp_bounce_message`,
+ - `mam_flush_messages`,
+ - `mam_muc_flush_messages`.
+
+## Metrics and instrumentation
+
+Metrics implementation has been reworked, and now a Prometheus endpoint is available.
+For more details, see [the instrumentation documentation](../configuration/instrumentation.md).
+This requires some changes to the configuration file, which are described in the following sections.
+
+### New `[instrumentation]` section
+
+To have metrics available in MongooseIM, an [`[instrumentation]` section](../configuration/instrumentation.md)  is now **required**.
+
+### `general` section
+
+`general.all_metrics_are_global` has been replaced by [`instrumentation.exometer.all_metrics_are_global`](../configuration/instrumentation.md#instrumentationexometerall_metrics_are_global).
+This option has no effect when using Prometheus, and should be removed from the config file in this case.
+
+### Exometer configuration changes
+
+Exometer is no longer configured through [`app.config`](../configuration/configuration-files.md#appconfig), but through [`mongooseim.toml`](../configuration/configuration-files.md#mongooseimtoml).
+
+Example configuration that is equivalent to the old example configuration provided in `app.config`:
+```toml
+[instrumentation]
+  probe_interval = 60000
+
+[[instrumentation.exometer.report.graphite]]
+  host = "127.0.0.1"
+  port = 2003
+  api_key = ""
+  interval = 60000
+  prefix = "mongooseim"
+  connect_timeout = 5000
+```
+
+See [Exometer configuration doc](../configuration/instrumentation.md#exometer-options) for more information.
+
+### New Prometheus endpoint
+
+Alternatively to Exometer, MongooseIM can now expose an endpoint serving Prometheus metrics.
+The Prometheus metrics provide labels, for example `host_type`.
+Because of this, the names of the metrics are usually shorter than the Exometer ones.
+
+The simplest configuration to enable it is:
```

Comment:
This only enables collecting metrics in MongooseIM, and doesn't enable the actual endpoint. I know it's clarified below, but maybe we could reword this?

Reply:
Right, I've changed it to include both sections as a proposed configuration.